Gordon360/Services/RecIM/SeriesService.cs

```diff
@@ -648,14 +731,120 @@ public async Task<EliminationRound> ScheduleElimRoundAsync(IEnumerable<SeriesTea
+                    TeamIDs = bye
+                });
+                matches.Add(createdMatch);
+                fullByeMatches.Add(0);
+                secondRoundMatches.Add(createdMatch.ID);
             }
+
+            // Bye pair matches, consisting of purely pairs of teams in the "second round" who both had byes, 3rd in order for bracket ordering
+            var byePairMatches = new List<int>();
+            foreach (var teamPair in byePairings)
+            {
+                var createdMatch = await _matchService.PostMatchAsync(new MatchUploadViewModel
+                {
+                    StartTime = series.StartDate, //default time, will be modified by autoscheduling
+                    SeriesID = series.ID,
+                    SurfaceID = 0, //default surface (undefined surface type) will be modified by autoscheduling
+                    TeamIDs = teamPair
+                });
+                matches.Add(createdMatch);
+                byePairMatches.Add(0);
+                byePairMatches.Add(0);
+                secondRoundMatches.Add(createdMatch.ID);
+            }
+
+            // Empty pairs, only happens if number of teams is greater than 24, where play-in matches outnumber bye matches. Last in order for bracket ordering.
+            foreach (var emptyMatchPair in secondRoundPlayInPairs)
+            {
+                var createdMatch = await _matchService.PostMatchAsync(new MatchUploadViewModel
+                {
+                    StartTime = series.StartDate, //default time, will be modified by autoscheduling
+                    SeriesID = series.ID,
+                    SurfaceID = 0, //default surface (undefined surface type) will be modified by autoscheduling
+                    TeamIDs = emptyMatchPair
+                });
+                matches.Add(createdMatch);
+                secondRoundMatches.Add(createdMatch.ID);
+            }
+
+            // assign bracket information
+            List<int> allMatches = fullByeMatches.Concat(byePairMatches.Concat(playInMatches)).ToList();
+
+            // bracket place first round
+            await CreateEliminationBracket(allMatches, 0);
+            // bracket place second round
+            await CreateEliminationBracket(secondRoundMatches, 1);
+
             return new EliminationRound
             {
-                TeamsInNextRound = teamPairings.Count() + numByes,
+                TeamsInNextRound = allMatches.Count()/2,
+                NumByeTeams = numByes,
                 Match = matches
             };
         }
 
+        private async Task<IEnumerable<MatchBracketViewModel>> CreateEliminationBracket(List<int> matchesIDs, int roundNumber)
+        {
+            var res = new List<MatchBracketViewModel>();
+            int rounds = (int)Math.Log(matchesIDs.Count(), 2)-1;
+            var matchArr = matchesIDs.ToArray();
+            var matchIndexes = new List<int> {  0, 1 };
```

Comment:
```suggestion
            var matchIndexes = new List<int> { 0, 1 };
```

Reply:
I don't know how you see this stuff but 👁️ 👁️ , very impressive and good catch!